packages/block-editor/src/store/actions.js

```diff
@@ -259,10 +259,15 @@ export function stopMultiSelect() {
 /**
  * Action that changes block multi-selection.
  *
- * @param {string} start First block of the multi selection.
- * @param {string} end   Last block of the multiselection.
+ * @param {string}      start                         First block of the multi selection.
+ * @param {string}      end                           Last block of the multiselection.
+ * @param {number|null} __experimentalInitialPosition Optional initial position. Pass as null to skip focus within editor canvas.
  */
-export const multiSelect = ( start, end ) => ( { select, dispatch } ) => {
+export const multiSelect = (
+	start,
+	end,
+	__experimentalInitialPosition = 0
```

Comment:
@andrewserong Why did we add this parameter? How does `initialPosition` make any sense for a multi block selection? `initialPosition` implies collapsed selection, while multi selection implies an uncollapsed selection.

Reply:
@ellatrix the existing `initialPosition` state is used in the writing flow to control whether or not to shift focus to the editor (for single block selection), so I thought it might make sense to re-use that same logic for multiple block selection so that the logic is as consistent as we can make it. In this case, we're using it effectively like a boolean, so setting to null says "there is no initial position" (and therefore don't focus in the editor canvas), and with the default `0` value, the position is "within the editor canvas".

For accessibility it was important that we don't shift focus to the editor canvas when multi-selecting from the list view. If there's a more semantic way to do this than using `initialPosition` we can definitely refactor it (one of the reasons I think @talldan suggested we use the `__experimental` prefix for the moment).

Do you have any guidance on how best to preserve the behaviour of skipping focusing on the editor canvas, if we were to refactor this?

Reply:
Yes, I understood the intent after digging a bit further. :) The naming is a bit confusing and I'm not sure if reusing the `0` value makes sense to mean selecting all content. Another thing to keep in mind is that focus and selection are two separate things. Even if focus has to stay in the List View, we still need to select the block contents. Anyway, it's not sure important right now :)

Reply:
Okay, great. Thanks for confirming @ellatrix, please do ping me if it does become important / needs updating, though, and I can follow up!